File: 228-summary-ranges.cpp
//https://leetcode.com/problems/summary-ranges
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if(nums.size() == 0)
            return {};
        vector<string> res;
        int start = nums[0];
        int prev = nums[0];
        for(int i=1; i<nums.size(); i++){
            if(nums[i] == (prev+1)){
                prev++;
            }else{
                if(start != prev){
                    string str = to_string(start) + "->" + to_string(prev);
                    res.push_back(str);
                }else{
                    string str = to_string(start);
                    res.push_back(str);
                }
                start = nums[i];
                prev = nums[i];
            }
        }

        if(start != prev){
            string str = to_string(start) + "->" + to_string(prev);
            res.push_back(str);
        }else{
            string str = to_string(start);
            res.push_back(str);
        }

        return res;
    }
};

// []
// [-2147483648,-2147483647,2147483647]
// [-2147483648,-2147483647,2147483647]
// [-1,2]
// [-1]
// [-5,-4,-2,-1,0,1,2]
// [0,2,3,4,6,8,9]
// [-1,0,2,9]

File: 228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        res = []
        if(len(nums) == 0):
            return res
        p = [str(nums[0]), None]
        for i in range(1, len(nums)):
            if(nums[i] - 1 == nums[i-1]):
                p[1] = str(nums[i])
            else:
                if(p[1] is None):
                    res.append(p[0])
                else:
                    res.append(f"{p[0]}->{p[1]}")
                p = [str(nums[i]), None]

        if(p[1] is None):
            res.append(p[0])
        else:
            res.append(f"{p[0]}->{p[1]}")
        
        return res
